app/src/main/java/com/testtask/apptesttask/presentation/characters/CharactersPresenter.kt

```diff
@@ -2,38 +2,36 @@ package com.testtask.apptesttask.presentation.characters
 
 import com.arellomobile.mvp.InjectViewState
 import com.testtask.apptesttask.model.interactor.characters.CharactersInteractor
+import com.testtask.apptesttask.model.system.SchedulersProvider
 import com.testtask.apptesttask.presentation.base.BasePresenter
-import io.reactivex.android.schedulers.AndroidSchedulers
-import io.reactivex.schedulers.Schedulers
 import javax.inject.Inject
 
 @InjectViewState
-class CharactersPresenter @Inject constructor(private val charactersInteractor: CharactersInteractor) :
-    BasePresenter<CharactersView>() {
+class CharactersPresenter @Inject constructor(
+    private var charactersInteractor: CharactersInteractor,
+    private var schedulersProvider: SchedulersProvider
+) : BasePresenter<CharactersView>() {
 
     override fun onFirstViewAttach() {
         super.onFirstViewAttach()
 
         loadCharacters()
     }
 
-    fun loadCharacters() {
-        unsubscribeOnDestroy(charactersInteractor.getCharacters()
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
+    private fun loadCharacters() {
+        charactersInteractor
+                .getCharacters()
+                .subscribeOn(schedulersProvider.io())
+                .observeOn(schedulersProvider.ui())
                 .doOnSubscribe { viewState.showProgress() }
-                .subscribe({ characterDataWrapper ->
-                    viewState.showCharacters(characterDataWrapper.data.results)
-                    hideProgress()
-                },
-                    { error ->
-                        viewState.showError(error as Throwable)
-                        hideProgress()
+                .doAfterTerminate { viewState.hideProgress() }
+                .subscribe(
+                    { characterDataWrapper ->
+                        viewState.showCharacters(characterDataWrapper.data.results)
+                    },
+                    { errorMessage ->
+                        viewState.showError(errorMessage.toString())
```

Comment:
Also you must create `ErrorHandler` to map errors into `String` message. See `GitFox` implementation.